code/modules/atmospherics/machinery/components/fusion/hfr_procs.dm

```diff
@@ -159,7 +159,20 @@
  * Getter for fusion fuel moles
  */
 /obj/machinery/atmospherics/components/unary/hypertorus/core/proc/check_fuel()
-	return (internal_fusion.gases[/datum/gas/tritium][MOLES] > FUSION_MOLE_THRESHOLD && internal_fusion.gases[/datum/gas/hydrogen][MOLES] > FUSION_MOLE_THRESHOLD)
+	if(!selected_fuel)
+		return FALSE
+	if(!internal_fusion.total_moles())
+		return FALSE
+	var/gas_check = 0
+	for(var/gas_type in selected_fuel.requirements)
+		internal_fusion.assert_gas(gas_type)
+		if(gas_type == selected_fuel.requirements[3])
```

Comment:
???????????????

Reply:
I have to check if the fuel mix exist in the internal gasmix, but the recipe requirement has 3 gases and only the first two are the actual fuel, the third is not but is going to be present inside the internal mix, so I have to weed it out

Reply:
If it's more common to reference fuel inputs and byproducts separately than to use them together, perhaps fuel inputs and fuel byproducts would be better as distinct variables?